Tools/CMake/AMReXUtils.cmake

```diff
@@ -48,6 +52,27 @@ function (get_amrex_version)
    set( AMREX_PKG_VERSION "${_pkg_version}" CACHE INTERNAL "" )
    unset(_pkg_version)
 
+   if (AMREX_GIT_VERSION)
```

Comment:
For this path, it might be good to check that this does not error out if the `AMREX_GIT_VERSION` returns `YY.MM` without any `-`.

Escaping looks good to me, but maybe some functions need `if` guards.
`string(FIND` will probably return `-1` in that case (https://cmake.org/cmake/help/latest/command/string.html#find), but `string(SUBSTRING` might count from the back (https://cmake.org/cmake/help/latest/command/string.html#substring), etc.

Reply:
On the other hand, I see that we use the same logic already above? ^^